operator/controllers/resources/role_controller_test.go

```diff
@@ -73,6 +75,62 @@ func TestRoleCreation(t *testing.T) {
 	})
 }
 
+func TestRoleCreationFromYAML(t *testing.T) {
+	ctx := context.Background()
+
+	teleportServer, operatorName := defaultTeleportServiceConfig(t)
+
+	require.NoError(t, teleportServer.Start())
+
+	tClient := clientForTeleport(t, teleportServer, operatorName)
+	k8sClient := startKubernetesOperator(t, tClient)
+
+	namespace := createNamespaceForTest(t, k8sClient)
+	roleName := "interns"
+
+	roleYAML := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer([]byte(`
+kind: role
+version: v5
+metadata:
+  name: interns
+spec:
+  allow:
+    node_labels:
+      '*': ['*']
```

Comment:
Is `'*': '*'` also OK? Mind adding a test for it?

Also, since we are here, let's test something more reasonable too, like two regular labels and multiple labels in the array.